pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/ClientCachedDao.java

```diff
@@ -0,0 +1,119 @@
+package org.broadinstitute.ddp.db.dao;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.List;
+
+import javax.cache.Cache;
+import javax.cache.expiry.Duration;
+
+import org.broadinstitute.ddp.cache.CacheService;
+import org.broadinstitute.ddp.db.dto.ClientDto;
+import org.broadinstitute.ddp.security.StudyClientConfiguration;
+import org.jdbi.v3.core.Handle;
+
+public class ClientCachedDao extends SQLObjectWrapper<ClientDao> implements ClientDao {
+    private static final String CACHE_NAME = "StudyClientConfigCache.byStudyClientConfigCacheKey";
```

Comment:
Implements ClientDao so this meant to be a drop in replacement for it.
Extends SQLObjectWrapper to pickup some shared implementation code. That is below.